bin/cylc

```diff
@@ -14,643 +15,225 @@
 #
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
+CATEGORIES=('control' 'con' 'information' 'info' 'all' 'task' 'license' \
+            'GPL' 'admin' 'prep' 'preparation' 'hook' 'discovery' 'utility' \
+            'util')
+                
+HELP_OPTS=('help' '--help' '-h' 'h' '?')
+
+get_version() {
+    # If there is a version file present, use that
+    if [[ -f "${CYLC_HOME}/VERSION" ]]; then
+        CYLC_VERSION="$(cat ${CYLC_HOME}/VERSION)"
+    else
+        local DESC=
+        DESC="$(cd ${CYLC_HOME} && git describe --abbrev=4 2>/dev/null)"
+        CYLC_VERSION="${DESC}"
+    fi
+    # Otherwise, use git to get the version information.
+    # Append "-dirty" if there are uncommitted changes.
+    if [[ -n "$(cd ${CYLC_HOME} && git status \
+            --untracked-files=no --porcelain)" ]]; then
+        CYLC_VERSION="${CYLC_VERSION}-dirty"
+    fi
+}
```

Comment:
If we run now `cylc` using a tarball, e.g. no `VERSION` file and no `.git/` directory, `cylc` will fail at the `git describe` statement, but will give no error message. This was causing failures for site-only tests under `tests/cylc-submit/`.